Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp

```diff
@@ -537,13 +557,38 @@ namespace Terrain
         }
     }
 
+    void TerrainPhysicsColliderComponent::BuildSurfaceTagToMaterialIndexLookup()
+    {
+        auto materialList = GetMaterialList();
+
+        // Lock this *after* calling GetMaterialList() so that we don't have nested locks.
+        AZStd::unique_lock lock(m_stateMutex);
+
+        m_surfaceTagToMaterialIndexLookup.clear();
+
+        for (auto& mapping : m_configuration.m_surfaceMaterialMappings)
```

Comment:
Nit: Looks like this could be `const auto& mapping` since the mapping itself isn't being modified